sql/src/main/java/io/crate/planner/consumer/ManyTableConsumer.java

```diff
@@ -250,6 +250,7 @@ static TwoTableJoin buildTwoTableJoinTree(MultiSourceSelect mss) {
             currentTreeRelationNames.add(rightName);
 
             // process where clause
+            pushDownNoMatch(mss, leftRelation, rightRelation);
```

Comment:
Wouldn't it make sense to change `QuerySpec::subset`` to call into ``WhereClause::subset`` and there make sure that a NO_MATCH is always returned?